### PR TITLE
Updated SalesForce to replace dead link, add hardware support

### PR DESCRIPTION
**Pull request by jadchaar**
Updated SalesForce entry to replace the dead doc link with an updated version. Added hardware support.

### CODE REVIEW

File: _data/communication.yml [hunk 1]
```diff
@@ -162,7 +162,8 @@ websites:
       sms: Yes
       software: Yes
       email: Yes
-      doc: https://help.salesforce.com/HTViewHelpDoc?id=security_require_two_factor_authentication.htm
+      hardware: Yes
+      doc: https://help.salesforce.com/articleView?id=security_overview_2fa.htm&type=5
```
Comment (mxxcon) [inline _data/communication.yml]: Is there a better link for end-users rather than admins? Perhaps https://trust.salesforce.com/en/security/2fa/ ?

Comment (jadchaar) [reply _data/communication.yml]: Good point. My only problem with the article you linked is that it seems to only promote their proprietary application, rather than an authenticator app in general. Thoughts on that?

Comment (jadchaar) [reply _data/communication.yml]: The only other relevant link I can find is a developer page, but that is even more cryptic. A bunch of the others just redirect to the admin page.

Comment (mxxcon) [reply _data/communication.yml]: I personally don't have too much of a problem with them promoting it. People who know the difference will be able to figure it out on their own. Otherwise, it should simplify support issues for Salesforce.

Comment (jadchaar) [reply _data/communication.yml]: Hi @mxxcon I think I found a good one: https://developer.salesforce.com/docs/atlas.en-us.identityImplGuide.meta/identityImplGuide/add_time-based_token.htm

Comment (mxxcon) [reply _data/communication.yml]: this link looks better 👍

File: _data/communication.yml [hunk 2]
```diff
@@ -162,7 +162,8 @@ websites:
       sms: Yes
       software: Yes
       email: Yes
-      doc: https://help.salesforce.com/HTViewHelpDoc?id=security_require_two_factor_authentication.htm
+      hardware: Yes
```
Comment (mxxcon) [inline _data/communication.yml:165]: wait a sec, why did you add hardware? we consider mobile authenticator apps to still be "software".
"hardware" are physical fobs or yubikey and similar type devices

Comment (Carlgo11) [reply _data/communication.yml:165]: They support U2F

**Review by mxxcon — APPROVED** [review]

**Review by Carlgo11 — APPROVED** [review]

File: _data/communication.yml (hunk 2 above)
Comment (jadchaar) [reply _data/communication.yml:165]: Yup I saw on that admin page that they supported U2F :)
